test/bin/scenario.sh

```diff
@@ -580,6 +580,32 @@ EOF
         "$@"
 }
 
+# Public function to run the tests for an offline scenario.  Greenboot checks should be handled
+# in RF suite setup, using the qemu-guest-agent.py library.
+run_offline_tests(){
+    local -r vmname="${1}"
+    local -r full_vmname="$(full_vm_name "${vmname}")"
+    shift
+
+    if [ ! -d "${RF_VENV}" ]; then
+        error "RF_VENV (${RF_VENV}) does not exist, create it with: ${ROOTDIR}/scripts/fetch_tools.sh robotframework"
+        exit 1
+    fi
+    local rf_binary="${RF_VENV}/bin/robot"
+    if [ ! -f "${rf_binary}" ]; then
+        error "robot is not installed to ${rf_binary}"
+        exit 1
+    fi
+
+    "${rf_binary}" \
+        --name "${SCENARIO}" \
+        --randomize all \
+        --loglevel TRACE \
+        --outputdir "${SCENARIO_INFO_DIR}/${SCENARIO}" \
+        -x junit.xml \
+        "$@"
+}
+
```

Comment:
This isn't exactly the place but: I don't think we'll get sos report for the test, right? Any plans to include it?

Reply:
This is true. It'll take some additional logic to collect it without the network.  I don't think it'd be difficult.  I agree it's not in the scope of this pr.